I_budget_verbessern

joeymez opened the pull request:
Fertigstellung des Items i_budget_verbessern:
                                                                                
Layout
- Anpassung des neuen Layouts in die neue Activity

Button
- Hinzufgen eines Buttons zum speichern
- Beim drücken werden die Inhalte an die nächste Activity weiter gegeben (A_Budget)

Calender
- Hinzufügen eines Kalenders für die Zeitspanne in welches das Budget geplant ist
- weiteres bitte aus Kommentare entnehmen

Logik
- Bevor gespeichert wird, werden alle EditTexts auf vollständigkeit geprüft - ansonsten erfolgt keine Speicherung
- Wenn alle Eingaben korrekt sind werden diese an die nächste Activity weitergegeben (ändert sich wenn DB vorhanden ist)
- Überprüfung ob eingegebener Betrag nicht zu hoch ist - wenn zu hoch wird darauf hingewiesen und nicht gespeichert

HINWEIS: 
- Wunsch zur Änderung des Kreises im Budget konnte diesmal leider nicht berücksichtigt werden. Erfolgt im nächsten Sprint ...
- Clickbare CardView ebenfalls ins nächste Spring
- Anbindung und Auslesung der Daten aus der DB erfolgt ebenfalls im nächsten Sprint

Commit:
Update Item i_budget_verbessern:

- Erste Änderung vom Layout der Budgetierung (lediglich ein Test)

Commit:
Update Item i_budget_verbessern:

- Zweite Änderung vom Layout der Budgetierung (neue Inhalte & Datum hinzugefügt)

Commit:
Update Item i_budget_verbessern:

- Hinzufügen einer neuen Activity zum Eingeben der Inhalte (statt des AlertDialog)
- Formatierung des Layouts für die neue Activity
- Erste Logiken implementiert (z.B. Start der neuen Activity beim drücken des FAB)

Commit:
Update Item i_budget_verbessern:

Layout
- Anpassung des neuen Layouts in die neue Activity

Button
- Hinzufgen eines Buttons zum speichern
- Beim drücken werden die Inhalte an die nächste Activity weiter gegeben (A_Budget)

Calender
- Hinzufügen eines Kalenders für die Zeitspanne in welches das Budget geplant ist
- weiteres bitte aus Kommentare entnehmen

Logik
- Bevor gespeichert wird, werden alle EditTexts auf vollständigkeit geprüft - ansonsten erfolgt keine Speicherung
- Wenn alle Eingaben korrekt sind werden diese an die nächste Activity weitergegeben (ändert sich wenn DB vorhanden ist)
- Überprüfung ob eingegebener Betrag nicht zu hoch ist - wenn zu hoch wird darauf hingewiesen und nicht gespeichert

HINWEIS:
- Wunsch zur Änderung des Kreises im Budget konnte diesmal leider nicht berücksichtigt werden. Erfolgt im nächsten Sprint ...
- Clickbare CardView ebenfalls ins nächste Spring
- Anbindung und Auslesung der Daten aus der DB erfolgt ebenfalls im nächsten Sprint

UliVe07 reviewed (changes requested):
Das Eingeben funktioniert bzw. wenn etwas fehlt kommt ein Fehler. 
Nur kommt bei mir am Handy kein Kalender zum Auswählen. 
Bei mehreren erstellten Budgets wird immer das vorherige überschrieben. 
Ansonsten schaut das Layout gut aus.

Commit:
Update Item i_budget_verbessern:

- Bug fix der Inhalte (werden nun alle Rot wenn sie leer sind)
- Anpassung der größte des Buttons

joeymez commented:
Hi @UliVe07 danke für dein Review! Ich kann dir  deine Kommentare beantworten:

- Bezüglich des Kalenders müsstest du ein Doppelklick machen damit es sich ändert da es eine EditText ist
 (Ich habe es jedoch nun auf eine TextView gemacht, sodass es beim ersten Klick aufgeht)

- Bezüglich der CardView selbst werden die eingegebenen Cards nicht gespeichert. Das liegt daran, dass die DB noch existiert und die eingaben somit dynamisch und nur Testdaten sind. Daher nicht wundern wenn Sie nicht da sind. Ich könnte eine Test-DB bauen aber das wäre den Aufwand nicht wert. D.h. dieser "BUG" ist kein Bug sondern lediglich eine kurze Sicht auf das Ergebnis was nach der Eingabe erfolgen soll :)

Commit:
Update Item i_budget_verbessern:

- Calender-Tag auf den heutigen Tag setzen
- Eingegebenes Datum wird nun auch auf die CardView übertragen

Commit:
Update Item i_budget_verbessern:

- verkürzung einiger Codeteile (rausstreichen einiger Funktionen / Methoden)
- Hinzufügen neuer Attribute für die dazugehörigen Bons für das Budget (Array liste & ID)
- Getter & Setter hinzugefügt für die neuen Attribute
- EditText für den Betrag begrenzt (auf 6 stellen)

Commit:
Update Item i_budget_verbessern:

- XML-Formatierung EDIT TEXT von NumberDecimal zu Number
- setzten von festen eingabe inhalte (0 - 9)

Sippl reviewed (approved):
Super sieht gut aus, Den erwähnten Bug mit den Punkt eingeben beim Budget hinzufügen hast ja bereits behoben :). Kleinere Optische Sachen können wir ja morgen diskutieren. Super :)
